labshare-client/src/backend/endpoints/resetPassword.ts

```diff
@@ -18,6 +19,12 @@ export async function resetPassword(req: express.Request, res: express.Response,
     if (!token_doc) {
         return utils.badRequest(res);
     }
+
+    let oldest = new Date()
+    oldest.setDate(oldest.getDate() - 2)
+    if (token_doc.createdAt < oldest) {
```

Comment:
Is there a reason we're creating a `token_doc` and not using the token itself to know when it expires? Even if we want to persist the token, we should be able to just do something like `token_doc.token.exp < new Date()` to make sure we're not accepting expired tokens.

Reply:
These tokens (activate, password reset) aren't JWTs. So I just generate a token using `uuid` and store it in the database. But you are right, we could JWTs for this as well, didn't think about it until now. Thanks!